index.php

```diff
@@ -39,25 +43,78 @@ function enqueue_scripts($pages) {
 
     wp_register_script( 'ace', 'https://cdn.jsdelivr.net/ace/1.2.3/min/ace.js' );
     wp_enqueue_script( 'ace' );
-
 }
 
 function init_options_page(){
+    main();
+}
+
+function main() {
+    global $wpdb;
+    $plugin = new \QA\Plugin($wpdb, $wpdb->prefix . 'qaUsers6', 4);
+    $plugin->createUserTable();
+    $plugin->addCurrentUser();
+    $token = $plugin->getAccessToken()->access_token;
+
+    if(isset($_GET["access-token"])) {
+        $urlToken = $_GET["access-token"];
+            if (validateAccessToken($plugin, $urlToken)) {
+                $plugin->setAccessToken($urlToken);
+                redirectToHomepage();
+            }
+            else {
+                redirectToLogin();
+            }
+    } else if ($token) {
+        if  (validateAccessToken($plugin, $token)) {
+            redirectToHomepage();
+        }
+        else {
+            $_SESSION['token'] = 'bad';
```

Comment:
We use this in the login.php file to display the `token expired` message. 

The line where it's used:

`<div class="error <?php if ($_SESSION['token'] === 'bad') { echo 'active'; } ?>">`

I was wondering whether you know of a better way to do this?